### PR TITLE
OAuth2: add domain attribute to token sessions

### PR DESCRIPTION
**Pull request by zhaohuabing**
Commit Message: 
Currently, the OAuth2 filter stores ID and access tokens in cookies without specifying a domain. This PR introduces support for specifying a domain for the cookies. With this addition, a domain can be configured, allowing users to remain logged in across all subdomains without needing to log in separately for each one.

Additional Description:
Risk Level: low
Testing: unit test
Docs Changes: API docs
Release Notes: Yes
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] implement: https://github.com/envoyproxy/envoy/issues/35490
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

Envoy Gateway issue: https://github.com/envoyproxy/gateway/issues/3919

### CODE REVIEW

File: api/envoy/extensions/filters/http/oauth2/v3/oauth.proto [hunk 1]
```diff
@@ -70,6 +70,10 @@ message OAuth2Credentials {
 
   // The cookie names used in OAuth filters flow.
   CookieNames cookie_names = 4;
+
+  // The domain to set the cookie on. If not set, the cookie will be set on the domain of the request.
```
Comment (htuch) [inline api/envoy/extensions/filters/http/oauth2/v3/oauth.proto:74]: If the unset behavior the same as existing behavior?

Comment (zhaohuabing) [reply api/envoy/extensions/filters/http/oauth2/v3/oauth.proto]: Yes, if unset the behavior doesn't change, defaulting to the host of the request, not including the subdomains.

**Review by htuch — COMMENTED** [review]
Let's have one of @derekargueta @mattklein123 take a look for implementation and SME coverage when this is ready for review (i.e. out of draft), but API LGTM.

**Comment by RyanTheOptimist** [comment]
@mattklein123 PTAL

**Comment by mattklein123** [comment]
LGTM. Can you merge main?

/wait

**Review by mattklein123 — APPROVED** [review]

**Comment by zhaohuabing** [comment]
Hi @RyanTheOptimist @htuch Is the failed check "envoy-presubmit" a must? I want to solve it, but couldn't find detailed error info in [the Azure pipeline](https://dev.azure.com/cncf/envoy/_build/results?buildId=177989&view=logs&j=76424b50-4542-51d2-ddfa-1b282263c744&t=d085f6bf-7203-5b28-a9c0-586a66155b4b).

**Comment by RyanTheOptimist** [comment]
> Hi @RyanTheOptimist @htuch Is the failed check "envoy-presubmit" a must? I want to resolve it, but couldn't find detailed error info in [the Azure pipeline](https://dev.azure.com/cncf/envoy/_build/results?buildId=177989&view=logs&j=76424b50-4542-51d2-ddfa-1b282263c744&t=d085f6bf-7203-5b28-a9c0-586a66155b4b).

Yes, it's definitely required. CI has been really unstable lately, alas. You might want/need to try doing a main merge and a re-push.

**Pull request merged**